packages/video_player/video_player_web/example/integration_test/pkg_web_tweaks.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@JS()
+library video_player_web_integration_test_pkg_web_tweaks;
+
+import 'dart:js_interop';
+import 'package:web/web.dart' as web;
+
+/// Adds a `controlsList` getter to `HTMLVideoElement`s.
+extension ControlsListInVideoElement on web.HTMLVideoElement {
+  external web.DOMTokenList? get controlsList;
+}
+
+/// Adds a `disablePictureInPicture` getter to `HTMLVideoElement`s.
+extension DisablePictureInPictureInVideoElement on web.HTMLVideoElement {
+  external JSBoolean get disablePictureInPicture;
+}
+
+/// Adds a `disableRemotePlayback` getter to `HTMLMediaElement`s.
+extension DisableRemotePlaybackInMediaElement on web.HTMLMediaElement {
+  external JSBoolean get disableRemotePlayback;
+}
+
+/// Retrieves the `Object` constructor from the DOM.
+@JS('Object')
+external DomObjectConstructor get jsObjectConstructor;
```

Comment:
@srujzs – best way to do this?

Reply:
Hah, this is pretty funny but it works because JavaScript.

Instead of this, you can just declare `defineProperty` and `_defineProperty` as static members and use `@JS('Object')` on top of `DomObjectConstructor` (and I'd rename `DomObjectConstructor` to something like `DomObject`). That better reflects the fact that `defineProperty` is an `Object` static member.

Reply:
The only reason I need this is because I want to have a version of a `video` element that returns `double.infinity` when getting its `duration`. I tried to implement that by wrapping a video element, but I was always getting `null` back?

I'll give this another shot.

Reply:
> I tried to implement that by wrapping a video element, but I was always getting null back?

Would be curious to see what the code looked like here. Considering the property is being declared as `writable`, I would assume an `external` setter would do the trick, but I could totally be missing context.

Reply:
Much better, also found an issue with the `seekTo` "noop" test which required expanding the `Descriptor` a little bit.